### PR TITLE
Implementing "Write your first thoughts" when creating a convo

### PR DESCRIPTION
**Pull request by psybuzz**
This PR addresses a missing feature that allows users to enter content into a "Write your thoughts" field when creating a conversation.  Upon creating the conversation, the backend will take the user's input and automatically create the first post using that text.

### CODE REVIEW

**Comment by tonyxj** [comment]
Looks like a pretty nice fix! LGTM. Deferring to @nikilselvam for a more in-depth look!

File: routes/conversation.js [hunk 1]
```diff
@@ -18,137 +19,161 @@ Post = db.models.Post;
...
-			    		// Send emails to notify participants of the new conversaion.
-						// This assumes that the 'username' field of a User is his/her email.
-						var peopleEmails = peopleData.map(function (person){
-							if (person) return person.username;
-						});
+					return d.promise;
+				});
+				Q.allSettled(jobs).then(function (){
+					// Redirect regardless of email success.
+			    	res.send({status: 'OK', success: true, redirect: '/conversation/'+conversation._id});
```
Comment (nikilselvam) [inline routes/conversation.js]: Extra tab!

File: routes/conversation.js [hunk 2]
```diff
@@ -18,137 +19,161 @@ Post = db.models.Post;
...
+
+		    		// Send emails to notify participants of the new conversaion.
```
Comment (nikilselvam) [inline routes/conversation.js]: Same here- extra tab will help this align with the following comment.

File: routes/post.js [hunk 3]
```diff
@@ -12,86 +12,122 @@ function resError(res, message){
...
-							var subject = 'GC: ' + currentUser.firstName + ' responded to the question: ' + convo.question;
-							if (subject.length > 70){
-								subject = subject.slice(0, 70) + '...';
-							}
-							var contentSnippet = content.slice(0, 140) + '...';
+		// Add the post to the conversation's discussion and save
+	    Conversation.findOne( { _id: conversationId }, function (err, convo, count){
+	    	if (err || !convo){
+	    		return onFailure("Could not find messages collection in DB.");
+			}
```
Comment (nikilselvam) [inline routes/post.js]: This parenthesis can be moved back one tab, I think, to match up with the rest of the if statement.

File: routes/post.js [hunk 4]
```diff
@@ -12,86 +12,122 @@ function resError(res, message){
...
+						console.log('New conversation notification emails have been sent.');
 					}
-			    });
-			}
-		});
-	}
-};
+				});
+			});
+
```
Comment (nikilselvam) [inline routes/post.js]: If I'm not mistaken, we might need a closing parenthesis here. It seems like there's one skipped `)};` sequence at the end of this function right now.

**Comment by nikilselvam** [comment]
Nice! I found a couple of semantic fixes, but nothing major with the logic. I'm not too familiar with how promises work still, but from what I can tell this looks pretty good. I'm also not on my computer currently and haven't had a chance to try this one out. @psybuzz have you tried creating a new conversation with the initial thoughts included? If so and it works, I'd say that once the little semantic issues are fixed this LGTM!

Also I like the placeholder change from `Topic` to `Question` for creating new conversations. I think that feels more intuitive

**Comment by psybuzz** [comment]
Okay, it's been awhile since the last commit, so I've addressed your changes @nikilselvam as well as a bug I found when re-testing the functionality.  The indentation should be better now, and I think the `});`'s you mentioned about should be all lined up.  I've tested it, and first-posts show up correctly without any errors!  Let me know what you think!

**Comment by nikilselvam** [comment]
Great! The changes look good to me, Erik. LGTM

**Pull request merged**